adagucserverEC/CAutoResource.cpp

```diff
@@ -40,23 +46,44 @@ int CAutoResource::configureDataset(CServerParams *srvParam,bool plain){
     internalDatasetLocation.replaceSelf(":","_");
     internalDatasetLocation.replaceSelf("/","_");
     
-    CT::string datasetConfigFile = srvParam->cfg->Dataset[0]->attr.location.c_str();
+    CT::string datasetConfigFile;
+    #ifndef ADAGUC_USE_KDCMONGODB
+      datasetConfigFile = srvParam->cfg->Dataset[0]->attr.location.c_str();
+    
+      datasetConfigFile.printconcat("/%s.xml",internalDatasetLocation.c_str());
+    
+      //Check whether this config file exists.
+      struct stat stFileInfo;
+      int intStat;
+      intStat = stat(datasetConfigFile.c_str(),&stFileInfo);
+      CT::string cacheBuffer;
+      //The file exists, so remove it.
+      if(intStat != 0) {
+        CDBDebug("Dataset config file does not exist: %s ",datasetConfigFile.c_str());
+        CDBError("No such dataset");
+        return 1;
+      }
+    #endif
+    #ifdef ADAGUC_USE_KDCMONGODB
+      int lastIndexOfUnderscore = internalDatasetLocation.lastIndexOf("_");
     
-    datasetConfigFile.printconcat("/%s.xml",internalDatasetLocation.c_str());
+      CT::string tmp_datasetname = internalDatasetLocation;
+      CT::string tmp_datasetversion = internalDatasetLocation;
     
-    CDBDebug("Found dataset %s",datasetConfigFile.c_str());
+      tmp_datasetname.substringSelf(0, lastIndexOfUnderscore);
+      tmp_datasetversion.substringSelf(lastIndexOfUnderscore + 1, internalDatasetLocation.length());
     
-    //Check whether this config file exists.
-    struct stat stFileInfo;
-    int intStat;
-    intStat = stat(datasetConfigFile.c_str(),&stFileInfo);
-    CT::string cacheBuffer;
-    //The file exists, so remove it.
-    if(intStat != 0) {
-      CDBDebug("Dataset config file does not exist: %s ",datasetConfigFile.c_str());
-      CDBError("No such dataset");
-      return 1;
-    }
+      CDBAdapterMongoDB *mongoDB = new CDBAdapterMongoDB();
+      if (mongoDB == NULL) {
```

Comment:
Voor de mongoDBadapter een prima plan. Daarna misschien voorstellen om dat meer te doen in de ADAGUC code. Misschien ook iets waar sonarqube iets over vindt?